bundles/org.openhab.binding.openthermgateway/src/main/java/org/openhab/binding/openthermgateway/handler/OpenThermGatewayHandler.java

```diff
@@ -0,0 +1,305 @@
+    public void connecting() {
+        connecting = true;
+        updateStatus(ThingStatus.OFFLINE);
+    }
+
+    @Override
+    public void connected() {
+        connecting = false;
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    @Override
+    public void disconnected() {
+        connecting = false;
+        try {
+            updateStatus(ThingStatus.OFFLINE);
+
+            // retry connection if disconnect is not explicitly requested
+            if (!explicitDisconnect) {
+                scheduler.schedule(new Runnable() {
+
+                    @Override
+                    public void run() {
+                        if (connector != null && !connecting && !connector.isConnected()) {
+                            connect();
+                        }
+                    }
+                }, config.connectionRetryInterval * 1000, TimeUnit.MILLISECONDS);
+            }
+        } catch (IllegalStateException ex) {
+        }
+    }
+
+    @Override
+    public void receiveMessage(Message message) {
+        if (DataItemGroup.dataItemGroups.containsKey(message.getID())) {
+            DataItem[] dataItems = DataItemGroup.dataItemGroups.get(message.getID());
+
+            for (int i = 0; i < dataItems.length; i++) {
+                DataItem dataItem = dataItems[i];
+
+                String channelId = dataItem.getSubject();
+
+                if (!OpenThermGatewayBindingConstants.SUPPORTED_CHANNEL_IDS.contains(channelId)) {
+                    continue;
+                }
+
+                State state = null;
+
+                switch (dataItem.getDataType()) {
+                    case Flags:
+                        state = TypeConverter.toOnOffType(message.getBit(dataItem.getByteType(), dataItem.getBitPos()));
+                        break;
+                    case Uint8:
+                    case Uint16:
+                        state = TypeConverter.toDecimalType(message.getUInt(dataItem.getByteType()));
+                        break;
+                    case Int8:
+                    case Int16:
+                        state = TypeConverter.toDecimalType(message.getInt(dataItem.getByteType()));
+                        break;
+                    case Float:
+                        state = TypeConverter.toDecimalType(message.getFloat());
+                        break;
+                    case DoWToD:
+                        break;
+                }
+
+                if (state != null) {
+                    logger.debug("Received update for channel '{}': {}", channelId, state.toFullString());
+                    try {
+                        updateState(channelId, state);
+                    } catch (IllegalStateException e) {
+                        // Missing callback, possibly due to incorrect initialization. how to handle correctly ?
+                    }
+                }
+            }
+        }
+    }
+
+    @Override
+    public void log(LogLevel loglevel, String format, String arg) {
```

Comment:
You should not need that. Use a logger in the other class (also applies to the other log methods).